editoast/src/infra_cache/mod.rs

```diff
@@ -533,10 +536,38 @@ impl InfraCache {
         conn: &mut DbConnection,
         infra_caches: &'a DashMap<i64, InfraCache>,
         infra: &Infra,
+        valkey: &Arc<ValkeyClient>,
     ) -> Result<Ref<'a, i64, InfraCache>> {
         // Cache hit
-        if let Some(infra_cache) = infra_caches.get(&infra.id) {
-            return Ok(infra_cache);
+        if let Some(mut infra_cache_mut) = infra_caches.get_mut(&infra.id) {
+            if infra.version != infra_cache_mut.infra_version {
+                let osrd_version = get_app_version().unwrap_or_default();
+                let infra_id = infra.id;
+                let mut valkey_conn = valkey.get_connection().await?;
+
+                let cache_operations = valkey_conn
+                    .json_zrangebyscore(
+                        format!("infra_patches.{osrd_version}.{infra_id})"),
```

Comment:
This format string could be in a function to ensure sync with the function that inserts patches.